src/nvim/eval/typval.c

```diff
@@ -1142,13 +1145,37 @@ void callback_free(Callback *callback)
   case kCallbackPartial:
     partial_unref(callback->data.partial);
     break;
+  case kCallbackLua:
+    NLUA_CLEAR_REF(callback->data.luaref);
+    break;
   case kCallbackNone:
     break;
   }
   callback->type = kCallbackNone;
   callback->data.funcref = NULL;
 }
 
+/// Check if callback is freed
+bool callback_is_freed(Callback callback)
+{
+  switch (callback.type) {
+  case kCallbackFuncref:
+    return false;
+    break;
+  case kCallbackPartial:
+    return false;
+    break;
+  case kCallbackLua:
+    return callback.data.luaref == LUA_NOREF;
```

Comment:
why not set `calback.type = kCallbackNone` instead of `{.type = kCallbackLua, .data.luaref = LUA_NOREF}` when clearing? Then this function can be deleted, and we can just check `callback.type == kCallbackNone` at the callsite.

Reply:
Is this something that could be done in a follow-up PR? (Or would that be too intrusive then?)

It would be good to get this merged sooner rather than later so it can start getting a proper beating from the nightly crowd...